src/main/java/com/greenfoxacademy/moviedatabaseapi/controller/CompanyController.java

```diff
@@ -16,13 +22,38 @@
     private String apiKey;
 
     private final CompanyService companyService;
+    private final AuthenticationManager authenticationManager;
+    private final MyUserDetailsService userDetailsService;
+    private final JwtUtil jwtUtil;
 
-    public CompanyController(CompanyService companyService) {
+    public CompanyController(CompanyService companyService, AuthenticationManager authenticationManager, MyUserDetailsService userDetailsService, JwtUtil jwtUtil) {
         this.companyService = companyService;
+        this.authenticationManager = authenticationManager;
+        this.userDetailsService = userDetailsService;
+        this.jwtUtil = jwtUtil;
     }
 
     @GetMapping("/company/{companyId}")
     public CompanyDTO getCompanies(@PathVariable Long companyId) {
         return companyService.getCompanies(companyId, apiKey);
     }
+
+    @RequestMapping(value = "/authenticate", method = RequestMethod.POST)
+    public ResponseEntity<?> createAuthenticationToken(@RequestBody AuthenticationRequest authenticationRequest) throws Exception {
```

Comment:
please create a separate service for authentication/authorization, e.g. AuthenticationService. In addition, an appropriate AuthenticationController can be added, so you will not have 
```
return ResponseEntity.ok(new AuthenticationResponse(jwt));
```
in your service, but in your controller